go/solution-operator/config/config_test.go

```diff
@@ -0,0 +1,174 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLoadConfigurationFromFile(t *testing.T) {
+	tests := map[string]struct {
+		filename string
+		result   *OperatorConfig
+		error    string
+	}{
+		"valid-online": {
+			filename: "online-config.yaml",
+			result: &OperatorConfig{
+				APIVersion: expectedAPIVersion,
+				Kind:       expectedKind,
+				Repositories: map[string][]Repository{
+					"1.0.0": []Repository{
+						Repository{
+							Endpoint: "docker.io/grafana",
+							Images:   []string{"grafana:6.7.4"},
+						},
+						Repository{
+							Endpoint: "docker.io/prom",
+							Images:   []string{"prometheus:v2.16.0"},
+						},
+						Repository{
+							Endpoint: "docker.io/bitnami",
+							Images:   []string{"prometheus-operator:v0.38.1"},
+						},
+					},
+				},
+			},
+			error: "",
+		},
+		"valid-metalk8s": {
+			filename: "metalk8s-config.yaml",
+			result: &OperatorConfig{
+				APIVersion: expectedAPIVersion,
+				Kind:       expectedKind,
+				Repositories: map[string][]Repository{
+					"1.0.0": []Repository{
+						Repository{
+							Endpoint: "metalk8s-registry-from-config.invalid/my-solution-1.0.0",
+							Images: []string{
+								"grafana:6.7.4",
+								"prometheus:v2.16.0",
+								"prometheus-operator:v0.38.1",
+							},
+						},
+					},
+					"1.1.0": []Repository{
+						Repository{
+							Endpoint: "metalk8s-registry-from-config.invalid/my-solution-1.1.0",
+							Images: []string{
+								"grafana:7.2.1",
+								"prometheus:v2.21.0",
+								"prometheus-operator:v0.42.1",
+							},
+						},
+					},
+				},
+			},
+			error: "",
+		},
+		"invalid-no-apiversion": {
+			filename: "invalid-no-apiversion.yaml",
+			result:   nil,
+			error:    "missing required field: apiVersion",
```

Comment:
Still new to go but the way we have been testing errors in zenko-operator has been nice. Instead of using the string of the error, we have been creating error values and using github.com/pkg/errors wrap method to add extra values. Then errors.cause to check the root error. Avoids having to copy paste strings and allows you to change error messages without having to also change tests just for a string value.

Reply:
For reference, https://blog.golang.org/go1.13-errors

Reply:
Thanks for the suggestion, I'll have a look!

Reply:
So I'll go with the "plain" Go 1.13 `Unwraps` as suggested in Nicolas' link, which should achieve the same effect as `github.com/pkg/errors` without requiring any additional package.

Reply:
@miniscruff I explained a bit more the choices around error handling in [`9b6d7b5` (#2844)](https://github.com/scality/metalk8s/pull/2844/commits/9b6d7b5e834cc540c6765d5d387f46f9cdec6360), which doesn't really match what is done for Zenko Operator.
In any case, thanks for pointing out that improvements could be made, I learnt a great deal while playing with all this!